store/sqlstore/user_store.go

```diff
@@ -2040,3 +2040,30 @@ func (us SqlUserStore) IsEmpty(excludeBots bool) (bool, error) {
 	}
 	return !hasRows, nil
 }
+
+func (us SqlUserStore) GetParticipantProfilesByIds(ctx context.Context, userIds []string, extended bool, allowFromCache bool) ([]*model.User, error) {
+	var users []*model.User
+
+	if extended {
+		query := us.usersQuery.
+			Where(map[string]interface{}{
+				"u.Id": userIds,
+			})
+
+		queryString, args, err := query.ToSql()
+
+		if err != nil {
+			return nil, errors.Wrap(err, "get_participant_profiles_by_ids_tosql")
+		}
+
+		if _, err := us.SqlStore.DBFromContext(ctx).Select(&users, queryString, args...); err != nil {
+			return nil, errors.Wrap(err, "failed to find Users")
+		}
+	} else {
+		for _, userId := range userIds {
+			users = append(users, &model.User{Id: userId})
+		}
+	}
```

Comment:
Trying to understand the reasoning behind adding this flag and doing this logic in the store layer. All this is doing is creating a clone of the slice only with the id field populated. Any reason we cannot do that in the app layer? That should remove the flag and make this method more straighforward.

Reply:
@agnivade, you are correct, also on a second thought this method is unnecessary, I'll refactor and use the `GetProfileByIds`.